tools/version-tracker/pkg/ecrpublic/ecrpublic.go

```diff
@@ -45,8 +46,13 @@ func GetLatestRevision(imageRepository, currentRevision string) (string, bool, e
 		}
 
 		if tagSemver.GreaterThan(latestRevisionSemver) {
-			latestRevisionSemver = tagSemver
-			latestRevision = tag
+			if branchName == constants.MainBranchName {
```

Comment:
Let's re-use the current logic we have for fetching latest GitHub releases.

Running it locally I got the expected results for main and non-main branches
```console
➜  bin/version-tracker display --project cilium/cilium 
ORGANIZATION  REPOSITORY  CURRENT VERSION  LATEST VERSION   
cilium        cilium      v1.13.18-eksa.1  v1.14.12-eksa.2                                                                /2.7s
➜  BRANCH_NAME=release-0.19 bin/version-tracker display --project cilium/cilium
ORGANIZATION  REPOSITORY  CURRENT VERSION  LATEST VERSION   
cilium        cilium      v1.13.18-eksa.1  v1.13.19-eksa.1
```